pkg/client/client.go

```diff
@@ -148,6 +150,19 @@ func NewClient(cfg *config.DataSetConfig, client *http.Client, logger *zap.Logge
 		firstReceivedAt:                 atomic.Int64{},
 		lastAcceptedAt:                  atomic.Int64{},
 		addEventsEndpointUrl:            addEventsEndpointUrl,
+		userAgent: fmt.Sprintf(
```

Comment:
Since this client library could be used outside of the context of the dataset exporter otel collector plugin, we should make this line more generic and only include dataset go library version and runtime version.

Other parts - otel collector information and dataset exporter information should be added by the library consumer - in this case, dataset exporter otel collector plugin.

So this plugin should define "base" User-Agent string and then other consumers of the library can provide optional suffix / similar which is added to this base User-Agent value. We use similar pattern in the agent - agent defined base user agent string and then each agent monitor adds optional suffix (we call it a fragment) to that base string which includes additional information about the monitor.

Reply:
Sound reasonable